### ISSUE TITLE
[Editor] On a Lenovo X13s Arm64 Windows laptop, clicking any element in certain popups will cause the body contents of the popup to visibly shift but not the interactable controls.

### ISSUE THREAD

**Issue by ggppjj** [godotengine/godot#96037]
### Tested versions

- Reproducible in 4.3 Stable Mono at least, and all other versions that I've tried including all dev builds of 4.3 and builds out to 4.1, both native Arm64 and x86_64 builds.

### System information

Godot v4.3.stable.mono - Windows 10.0.27686 - GLES3 (Compatibility) - D3D12 (Qualcomm(R) Adreno(TM) 8cx Gen 3) - Snapdragon (TM) 8cx Gen 3 @ 3.0 GHz (8 Threads)

### Issue description

When clicking on the inner body of some windows in Godot on this laptop, the contents will visually shift but the clickable controls will not. 

This is probably going to be a weird one, because it's a Windows on Arm device.

![image](https://github.com/user-attachments/assets/1857d091-9c1f-4644-9250-22d0ee722264)

### Steps to reproduce

1. Be on a Thinkpad X13s.
2. Open Godot.
3. Create a new project.
4. Click "Browse".
5. Click anywhere in the explorer window view.

### Minimal reproduction project (MRP)

N/A

**Comment by ggppjj** [comment]
Apologies for the out-of-the-blue ping, but I have seen @JustinBeckerMSFT act as a valuable resource for Windows on ARM issues on PrusaSlicer here on GitHub in the past and figure it would probably not hurt to introduce them to this issue.

**Comment by JustinBeckerMSFT** [comment]
@ggppjj Am I understanding your initial report correctly that this issue repros under emulation and natively? If so, I suspect it's not specifically related to ARM64. UI stuff is not my area of expertise, but my first guess would be something like DPI awareness. The X13s has a relatively high DPI display. 

As luck would have it, i'm writing this response on an X13s, so i'll try to reproduce this locally and report back :)

**Comment by ggppjj** [comment]
That's right, I wasn't even aware of a native Godot ARM64 build until recently, and had assumed it to be an emulation error until now. 

DPI awareness issues sound reasonable, the only reason that I'm leaning more towards platform issues is because I would assume Godot to be used in many high-DPI environments on other hardware platforms without issue because of the platforms that the engine under the hood does support natively. This could instead be a Godot Windows issue more generally, but then I would also expect Windows users of Godot with HighDPI screens to be common enough as well. There is at least one HighDPI Windows open issue in the issue tracker at the moment, and I would personally be surprised if I was the first to have stumbled on a more generic issue, but that's just my hunch so far.

It seems like the common thread across the windows with issues are ones that minimize in what I can only relate as a "Windows 3.11-style". They don't minimize to the taskbar, they minimize to become a tiny titlebar and go to the bottom left of the taskbar.

**Comment by ggppjj** [comment]
@JustinBeckerMSFT Were you able to find time to attempt to reproduce this?

**Comment by JustinBeckerMSFT** [comment]
@ggppjj Thanks for the reminder! Yes, I was able to repro this issue on one of my machines. What's curious to me about this issue is that it seems to only be _some_ file open dialogues that are affected. I know very little about the Windows UI APIs so it's not clear to me if this is an issue on our side. What I will say is I tried to repro the issue at multiple display scaling values, and it seemed to repro consistently regardless of display scaling, so my original DPI scaling theory is probably bunk.
